Signal/src/ViewControllers/DebugUI/DebugUIMessagesAction.h

```diff
@@ -0,0 +1,62 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "DebugUIMessagesUtils.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface DebugUIMessagesAction : NSObject
```

Comment:
Introduces the concept of an "action".  

* "Single" actions typically create or send a single message.
* "Group" actions are used to:
   * Perform N actions drawn randomly from a group of subactions.
   * Perform all actions in a group exactly once.